src/test/net/SSLHTTPDownloader_test.cpp

```diff
@@ -0,0 +1,269 @@
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/net/SSLHTTPDownloader.h>
+#include <test/jtx.h>
+#include <test/jtx/TrustedPublisherServer.h>
+#include <test/unit_test/FileDirGuard.h>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <mutex>
+#include <condition_variable>
+
+namespace ripple {
+namespace test {
+
+class SSLHTTPDownloader_test : public beast::unit_test::suite
+{
+    // TODO return temporary with copy elision in c++17
+    std::unique_ptr<TrustedPublisherServer> createServer(
+        jtx::Env& env,
+        bool ssl = true)
+    {
+        std::vector<TrustedPublisherServer::Validator> list;
+        list.push_back (TrustedPublisherServer::randomValidator());
+        return std::make_unique<TrustedPublisherServer> (
+            env.app().getIOService(),
+            list,
+            env.timeKeeper().now() + std::chrono::seconds{3600},
+            ssl);
+    }
+
+
+    struct DownloadCompleter
+    {
+        std::mutex m;
+        std::condition_variable cv;
+        bool called = false;
+        boost::filesystem::path dest;
+
+        void operator ()(boost::filesystem::path dst)
+        {
+            std::unique_lock<std::mutex> lk(m);
+            called = true;
+            dest = std::move(dst);
+            lk.unlock();
+            cv.notify_one();
+        };
+
+        bool waitComplete()
+        {
+            std::unique_lock<std::mutex> lk(m);
+            using namespace std::chrono_literals;
+            return cv.wait_for(lk, 1s, [this]{return called;});
+        };
+    };
+
+    struct Downloader
+    {
+        test::StreamSink sink_;
+        beast::Journal journal_;
+        // The SSLHTTPDownloader must be created as shared_ptr
+        // because it uses shared_from_this
+        std::shared_ptr<SSLHTTPDownloader> ptr_;
+
+        Downloader(jtx::Env& env)
+        : journal_ {sink_}
+        , ptr_ {std::make_shared<SSLHTTPDownloader>(
+              env.app().getIOService(), journal_, env.app().config())}
+        {}
+
+        SSLHTTPDownloader* operator->()
+        {
+            return ptr_.get();
+        }
+    };
+
+
```

Comment:
nit: extra lf

Reply:
I'm not sure what this comment applies to